### PR TITLE
Add AvoidNegatedCollectionFilterOrRejectRector

### PR DESCRIPTION
**Pull request by spawnia**
The test currently fails and I am not sure how to make it work.

```diff
 (new Collection([0, 1, null, -1]))
     ->reject(fn (?int $number): bool => is_null($number))
-    ->filter(fn (int $number): bool => $number > 0);
+    ->reject(fn (int $number): bool => ! ($number > 0));
```

I think there can be additional logic to simplify conditionals that use negated comparisons:

```diff
- ->filter(fn (int $n): bool => $n !== 0)
+ ->reject(fn (int $n): bool => $n === 0)
```

I also wonder where this rule would be included, perhaps in the code quality set list?

What do you think about this idea in general? Would you be up for inclusion?

### CODE REVIEW

**Comment by GeniJaho** [comment]
@spawnia thanks for your PR. I think it would make a great addition to the Code Quality set. At first, it can be simple with just the negated expressions, not the negated comparisons. At least, until we have some feedback from users.
I will also try this on a few real-life projects and check for any edge cases.

We also need to support normal functions, not just arrow functions, so I would add a test for them as well. You can use `dump_node()` and `print_node()` functions to help you working with different node types. Also this can be helpful https://getrector.com/ast.

The test is failing because Rector does not know what the `Collection->filter()` method returns. We need to supply a stub in `stubs/Illuminate/Support/Collection.php`, check how the other stubs are implemented, you can copy the `filter()` and `reject()` methods' signatures from the real class in the Laravel framework. After you implement a stub, you need `composer dump-autoload` locally and the tests should be passing. If it's unclear :sweat_smile: I can fix that part in your PR.

**Comment by spawnia** [comment]
> We also need to support normal functions, not just arrow functions

I thought about this, but found that it can become really tricky. For example, what are we supposed to do if there are multiple return statements? I would leave it for now.

**Review by GeniJaho — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -12,42 +13,50 @@ This package is a [Rector](https://github.com/rectorphp/rector) extension develo
...
 |---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [LaravelSetList::LARAVEL_ARRAYACCESS_TO_METHOD_CALL](https://github.com/driftingly/rector-laravel/blob/main/config/sets/laravel-arrayaccess-to-method-call.php)                             | Converts uses of things like `$app['config']` to `$app->make('config')`.                                                                                                                                                                         |
```
[the remark this answers is not in the document]
Comment (spawnia) [reply README.md:43]: Done, I opened https://github.com/driftingly/rector-laravel/pull/237. I think I properly documented the differences between sets there, especially given that there is no documentation about this at all right now 😉 But let's discuss this further in the new pull request.

Thanks for your feedback on this!

**Review by GeniJaho — APPROVED** [review]
